lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py

```diff
@@ -801,130 +807,41 @@ def create_or_update_load_balancer(self, param):
-            etag=_.etag
-        ) for _ in load_balancer.backend_address_pools]
-
-    if load_balancer.load_balancing_rules:
-        result['load_balancing_rules'] = [dict(
-            id=_.id,
-            name=_.name,
-            protocol=_.protocol,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            backend_address_pool_id=_.backend_address_pool.id,
-            probe_id=_.probe.id,
-            load_distribution=_.load_distribution,
-            frontend_port=_.frontend_port,
-            backend_port=_.backend_port,
-            idle_timeout_in_minutes=_.idle_timeout_in_minutes,
-            enable_floating_ip=_.enable_floating_ip,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.load_balancing_rules]
-
-    if load_balancer.probes:
-        result['probes'] = [dict(
-            id=_.id,
-            name=_.name,
-            protocol=_.protocol,
-            port=_.port,
-            interval_in_seconds=_.interval_in_seconds,
-            number_of_probes=_.number_of_probes,
-            request_path=_.request_path,
-            provisioning_state=_.provisioning_state
-        ) for _ in load_balancer.probes]
-
-    if load_balancer.inbound_nat_rules:
-        result['inbound_nat_rules'] = [dict(
-            id=_.id,
-            name=_.name,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            protocol=_.protocol,
-            frontend_port=_.frontend_port,
-            backend_port=_.backend_port,
-            idle_timeout_in_minutes=_.idle_timeout_in_minutes,
-            enable_floating_point_ip=_.enable_floating_point_ip if hasattr(_, 'enable_floating_point_ip') else False,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.inbound_nat_rules]
-
-    if load_balancer.inbound_nat_pools:
-        result['inbound_nat_pools'] = [dict(
-            id=_.id,
-            name=_.name,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            protocol=_.protocol,
-            frontend_port_range_start=_.frontend_port_range_start,
-            frontend_port_range_end=_.frontend_port_range_end,
-            backend_port=_.backend_port,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.inbound_nat_pools]
-
-    if load_balancer.outbound_nat_rules:
-        result['outbound_nat_rules'] = [dict(
-            id=_.id,
-            name=_.name,
-            allocated_outbound_ports=_.allocated_outbound_ports,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            backend_address_pool=_.backend_address_pool.id,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.outbound_nat_rules]
-
-    return result
+def default_compare(new, old, path):
+    if new is None:
```

Comment:
what if old is  none too?

Reply:
you mean if old is not none?
at this point I am just assuming that if parameter is not specified (is None), then we don't do comparison. I am not sure whether it's right or wrong default behavior in idempotency check.

Reply:
then why check new not None?

Reply:
in this case we should just remove this check.... and then it will fall under else.

Reply:
actually i would suggest keep none check inside this function. as a function, you cannot have assumption on what will be passed in.  it's hard to rely  on caller to do none check.

Reply:
well, I just removed it, if it's None it will fall under else, and there will be just == comparison, which will return True if old is also None, or False if old is something else.